dot/state/block_finalisation.go

```diff
@@ -63,16 +63,6 @@ func (bs *BlockState) GetFinalisedHash(round, setID uint64) (common.Hash, error)
 }
 
 func (bs *BlockState) setHighestRoundAndSetID(round, setID uint64) error {
-	currRound, currSetID, err := bs.GetHighestRoundAndSetID()
-	if err != nil {
-		return err
-	}
-
-	// higher setID takes precedence over round
-	if setID < currSetID || setID == currSetID && round <= currRound {
```

Comment:
I know that for relaxed mode we don't need to verify the round number, but do we also now skip verifying that the setId is greater?